src/Components/Header.jsx

```diff
@@ -103,7 +110,7 @@ function Header(){
                   </Button>
                 </Search>
                 <div>
-                  {user ? (<Button onClick={logout} variant="">Logout</Button>) : <Button onClick={ login } variant="contained">Login</Button>}
+                  {user ? (<Button onClick={logout} variant="">Logout</Button>) : <Button onClick={ handleLogin } variant="contained">Login</Button>}
```

Comment:
you could pass the navigate straight inside the onClick for economy:
```suggestion
                  {user ? (<Button onClick={logout} variant="">Logout</Button>) : <Button onClick={ () => navigate("/login")) } variant="contained">Login</Button>}
```
But it's fine to keep the Handler